gym/wrappers/record_episode_statistics.py

```diff
@@ -1,39 +1,59 @@
 import time
 from collections import deque
+import numpy as np
 import gym
 
 
 class RecordEpisodeStatistics(gym.Wrapper):
     def __init__(self, env, deque_size=100):
         super(RecordEpisodeStatistics, self).__init__(env)
+        self.env_is_vec = True
+        if not isinstance(env, gym.vector.VectorEnv):
+            self.num_envs = 1
+            self.env_is_vec = False
         self.t0 = (
             time.time()
         )  # TODO: use perf_counter when gym removes Python 2 support
-        self.episode_return = 0.0
-        self.episode_length = 0
+        self.episode_count = 0
+        self.episode_returns = None
+        self.episode_lengths = None
         self.return_queue = deque(maxlen=deque_size)
         self.length_queue = deque(maxlen=deque_size)
 
     def reset(self, **kwargs):
-        observation = super(RecordEpisodeStatistics, self).reset(**kwargs)
-        self.episode_return = 0.0
-        self.episode_length = 0
-        return observation
+        observations = super(RecordEpisodeStatistics, self).reset(**kwargs)
+        self.episode_returns = np.zeros(self.num_envs, dtype=np.float32)
```

Comment:
`self.num_envs` is not defined here if `env` is a `VectorEnv` instance.

Reply:
I don’t follow. `VectorEnv` has a `num_envs` attribute, right?

Reply:
Oh right the wrapper inherits the properties from `env`, sorry my bad!